test/stat_test.py

```diff
@@ -33,3 +34,13 @@ def test_missing_path(self):
     def test_invalid_path(self):
         self.assertRaises(FileNotFoundException, self.client.stat,
                           ['/does/not/exist'])
+
+    def test_format_permission(self):
```

Comment:
For writing octal integer consider [pep-3127](https://www.python.org/dev/peps/pep-3127/) - so for example form 0o1777.